### ISSUE TITLE
Maven respository version number problems

### ISSUE THREAD

**Issue by eliasvasylenko** [Frege/frege#315]
So far as I understand, the version numbers `3.24.100.1-jdk7` and `3.24.100.1-jdk8` are not strictly valid by Maven standards. Instead the version is parsed to mean:

Major: 0
Minor: 0
Incremental: 0
Qualifier: 3.24.100.1-jdk8

They should probably instead have been deployed as something like `3.24.100-1-jdk7` and `3.24.100-1-jdk8`, which would parse as:

Major: 3
Minor: 24
Incremental: 100
Qualifier: 1-jdk8

Many tools cannot even resolve the artefacts at all (and I wouldn't expect them to). Is there any chance of redeployment with different version numbers?

**Comment by Ingo60** [comment]
Great idea.
Can you do that?

**Comment by talios** [comment]
On 7 Nov 2016, at 1:30, Elias N Vasylenko wrote:

> Major: 0
> Minor: 0
> Incremental: 0
> Qualifier: 3.24.100.1-jdk8
>
> They should probably instead have been deployed as something like `3.24.100-jdk7` and `3.24.100-jdk8`.

Not really, it's 3.24.100.1 with a classifier of jdk7/jdk8

Major: 3
Minor: 24
Incremental: 100.1
Classifier: jdk7

http://search.maven.org/#artifactdetails%7Corg.frege-lang%7Cfrege%7C3.24.100.1%7Cpom

The 100.1 was a point release after changing the deployment. Tho, I see 100 itself doesn't exist in central, so maybe I didn't need to do that ( I can't recall off hand ).






-- 
Mark Derricutt
http://www.theoryinpractice.net
http://www.chaliceofblood.net
http://plus.google.com/+MarkDerricutt
http://twitter.com/talios
http://facebook.com/mderricutt

**Comment by eliasvasylenko** [comment]
@talios I think the truth was somewhere in between.

First of all, I clearly have misunderstood the purpose of classifiers, I'm sorry about that. This solves the problem I was having, which is embarrassing.

That said, I think the bug is still *technically* valid, it's just not very important. I tested  according to the core Maven class `org.apache.maven.artifact.versioning.DefaultArtifactVersion` and it agrees with how I interpreted all the documentation I could find:

Major: 0
Minor: 0
Incremental: 0
Qualifier: 3.24.100.1

Does this seem agreeable to you? The numeric components of the standard version format are all integers, and the qualifier must be preceded by `-` not `.`.

Thankfully the Maven version ordering rules are completely independent of their standard version format rules, so version ranges still work as expected (rather than sorting on the qualifier as a string or something awful).

@Ingo60 I think it's clear I'm not qualified to make such a pull request ;) and I wouldn't have felt comfortable messing directly with someone else's project's versioning anyway.
